bugsnag-android-core/src/test/java/com/bugsnag/android/LaunchCrashDeliveryTest.kt

```diff
@@ -0,0 +1,136 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.BugsnagTestUtils.generateConfiguration
+import com.bugsnag.android.BugsnagTestUtils.generateEvent
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import java.lang.Thread
+import java.nio.file.Files
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * Verifies that crashes on launch are sent synchronously and other crashes are not.
+ */
+class LaunchCrashDeliveryTest {
+
+    private lateinit var storageDir: File
+    private lateinit var errorDir: File
+
+    @Before
+    fun setUp() {
+        storageDir = Files.createTempDirectory("tmp").toFile()
+        storageDir.deleteRecursively()
+        errorDir = File(storageDir, "bugsnag-errors")
+    }
+
+    @After
+    fun tearDown() {
+        storageDir.deleteRecursively()
+    }
+
+    /**
+     * Verifies that flushOnLaunch() blocks the main thread by simulating a slow delivery.
+     */
+    @Test
+    fun flushOnLaunchBlocksMainThread() {
+        val delivery = SlowDelivery()
+        val backgroundTaskService = BackgroundTaskService()
+        val eventStore = createEventStore(delivery, backgroundTaskService)
+        val event = generateEvent()
+        event.app.isLaunching = true
+        eventStore.write(event)
+
+        // check time difference in ms is >1500, proving thread was blocked
+        val baseline = System.currentTimeMillis()
+        eventStore.flushOnLaunch()
+        val now = System.currentTimeMillis()
+        assertTrue(now - baseline > 1500)
+        backgroundTaskService.shutdown()
+        assertEquals(1, delivery.count.get())
+    }
+
+    /**
+     * Crashes on launch are delivered synchronously
+     */
+    @Test
+    fun flushOnLaunchSync() {
```

Comment:
It looks like there's a missing test case here for generating two launch events and asserting that the correct one is delivered. Perhaps `TestDelivery` could keep the most recent `errorClass` sent or something to test against.

Reply:
Added an extra test case that asserts this.